cmake/scripts/cmake_easy_setup.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+import argparse
+import subprocess
+import shutil
+import tempfile
+import os
+import pathlib
+import json
+from jinja2 import Environment, FileSystemLoader
+import difflib
+
+from parse_boards import parse_file
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--cli", "-x", type=pathlib.Path, required=False, default=shutil.which("arduino-cli"), help="path to arduino-cli")
+parser.add_argument("--board", "-b", type=str, default="", help="board name")
+parser.add_argument("--fire", action="store_true", default=False, help="launch the build immediately (use with caution!)")
+output_args = parser.add_mutually_exclusive_group(required=True)
+output_args.add_argument("--output", "-o", type=pathlib.Path, help="output file (CMake) with placeholders")
+output_args.add_argument("--sketch", "-s", type=pathlib.Path, help="output file (CMake) filled given a sketch folder")
+
+shargs = parser.parse_args()
+shargs.board = shargs.board.upper()
+
+if shargs.sketch and not shargs.board :
+    print("""
+    Warning: you did not specify which board you were targeting;
+    please review the generated CMakeLists.txt to remove the placeholder value before calling `cmake`.
+    """)
+
+if shargs.cli is None :
+    print("""
+    Error: `arduino-cli` not found in $PATH.
+    Please install arduino-cli and make it available from your system $PATH,
+    or give its location through the `--cli` argument.
+    """)
+    exit(-1)
```

Comment:
Will be never called as argparse will raised automatically an error

Reply:
* about `shargs.sketch and not shargs.board` : both are optional arguments, so the condition _can_ fire.
Try running `python3 ./cmake/scripts/cmake_easy_setup.py -s ~/1_Blink/`...

* about `shargs.cli is None` : the default value is `shutil.which("arduino-cli")`, and that can be None: https://docs.python.org/3/library/shutil.html#shutil.which.